synapse/module_api/__init__.py

```diff
@@ -773,9 +773,9 @@ async def get_user_ip_and_agents(
             # Sanitize some of the data. We don't want to return tokens.
             return [
                 UserIpAndAgent(
-                    ip=str(data["ip"]),
-                    user_agent=str(data["user_agent"]),
-                    last_seen=int(data["last_seen"]),
+                    ip=data["ip"],
+                    user_agent=data["user_agent"],
+                    last_seen=data["last_seen"],
```

Comment:
Note to self: last_seen was `int`, is now a `str`?

Reply:
Could you explain why this is safe? (Maybe I just need to chase the logic through next week?)

Reply:
I thought `data["last_seen"]` is still an `int`?

`raw_data` used to be a `Dict[str, Union[str, int]]`, but is now a `TypedDict` so these ought to already have the correct type.